### PR TITLE
[ko] Update outdated files in `dev-1.23-ko.3` (M17-M43)

### PR DESCRIPTION
**Pull request by jihoon-seo**
한 PR에 너무 많은 변경사항을 포함시키면 리뷰하기 어려울 수 있어서, 적절히 분할하여 올립니다. 😊

- Related issues: 
  - #33473 (**dev-1.23-ko.3**)
  - #33472 (**dev-1.24-ko.1**)
- Related PRs: 
  - #33878
  - #33911
  - #33976

> ### 27 files to be modified
> * [ ]  M17.  `content/en/docs/concepts/containers/container-lifecycle-hooks.md`  | 13(+S) 13(-)
> * [ ]  M18.  `content/en/docs/concepts/containers/images.md`  | 12(+S) 97(-)
> * [ ]  M19.  `content/en/docs/concepts/containers/runtime-class.md`  | 1(+XS) 2(-)
> * [ ]  M20.  `content/en/docs/concepts/extend-kubernetes/_index.md`  | 2(+XS) 1(-)
> * [ ]  M21.  `content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md`  | 1(+XS) 1(-)
> * [ ]  M22.  `content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md`  | 2(+XS) 1(-)
> * [ ]  M23.  `content/en/docs/concepts/extend-kubernetes/operator.md`  | 1(+XS) 0(-)
> * [ ]  M24.  `content/en/docs/concepts/extend-kubernetes/service-catalog.md`  | 1(+XS) 0(-)
> * [ ]  M25.  `content/en/docs/concepts/overview/components.md`  | 2(+XS) 2(-)
> * [ ]  M26.  `content/en/docs/concepts/overview/kubernetes-api.md`  | 1(+XS) 1(-)
> * [ ]  M27.  `content/en/docs/concepts/overview/what-is-kubernetes.md`  | 2(+XS) 2(-)
> * [ ]  M28.  `content/en/docs/concepts/overview/working-with-objects/kubernetes-objects.md`  | 14(+S) 5(-)
> * [ ]  **M29.**  `content/en/docs/concepts/scheduling-eviction/api-eviction.md`  | **108(+L) 5(-)**
> * [ ]  **M30.**  `content/en/docs/concepts/scheduling-eviction/assign-pod-node.md`  | **252(+L) 212(-)**
> * [ ]  M31.  `content/en/docs/concepts/scheduling-eviction/pod-overhead.md`  | 2(+XS) 2(-)
> * [ ]  M32.  `content/en/docs/concepts/scheduling-eviction/resource-bin-packing.md`  | 1(+XS) 1(-)
> * [ ]  M33.  `content/en/docs/concepts/security/controlling-access.md`  | 6(+XS) 0(-)
> * [ ]  M34.  `content/en/docs/concepts/security/overview.md`  | 1(+XS) 1(-)
> * [ ]  M35.  `content/en/docs/concepts/services-networking/dns-pod-service.md`  | 11(+S) 0(-)
> * [ ]  M36.  `content/en/docs/concepts/services-networking/endpoint-slices.md`  | 3(+XS) 3(-)
> * [ ]  M37.  `content/en/docs/concepts/services-networking/ingress-controllers.md`  | 2(+XS) 1(-)
> * [ ]  M38.  `content/en/docs/concepts/services-networking/ingress.md`  | 15(+S) 9(-)
> * [ ]  M39.  `content/en/docs/concepts/services-networking/network-policies.md`  | 2(+XS) 37(-)
> * [ ]  M40.  `content/en/docs/concepts/services-networking/service.md`  | 43(+M) 10(-)
> * [ ]  M41.  `content/en/docs/concepts/services-networking/topology-aware-hints.md`  | 7(+XS) 1(-)
> * [ ]  M42.  `content/en/docs/concepts/storage/ephemeral-volumes.md`  | 15(+S) 8(-)
> * [ ]  M43.  `content/en/docs/concepts/storage/persistent-volumes.md`  | 1(+XS) 1(-)

[참고]
변경 사항 중에서 링크 업데이트를 반영하지 않은 부분이 있는데,
이는
- 'Debugging 섹션을 재구성하는 미래의 PR' 에서 처리될 예정이거나,
- #33473 의 R1-R9 를 처리하는 PR에서 
  링크도 함께 업데이트 하거나,
- #33473 가 마무리될 쯤에 
  `dev-1.23-ko.3` 의 링크들을 다시 한 번 점검하고 업데이트 하는 것이 좋을 것 같습니다.

/language ko

### CODE REVIEW

File: content/ko/docs/concepts/containers/container-lifecycle-hooks.md [hunk 1]
```diff
@@ -99,28 +99,28 @@ TERM 신호를 보내기 전에 실행을 완료해야 한다. 실행 중에 `Pr
...
+### 훅 핸들러 디버깅
 
 훅 핸들러의 로그는 파드 이벤트로 노출되지 않는다.
 만약 핸들러가 어떠한 이유로 실패하면, 핸들러는 이벤트를 방송한다.
 `PostStart`의 경우, 이것은 `FailedPostStartHook` 이벤트이며,
 `PreStop`의 경우, 이것은 `FailedPreStopHook` 이벤트이다.
-이 이벤트는 `kubectl describe pod <파드_이름>`를 실행하면 볼 수 있다.
-다음은 이 커맨드 실행을 통한 이벤트 출력의 몇 가지 예다.
+실패한 `FailedPreStopHook` 이벤트를 직접 생성하려면, [lifecycle-events.yaml](https://raw.githubusercontent.com/kubernetes/website/main/content/en/examples/pods/lifecycle-events.yaml) 파일을 수정하여 postStart 명령을 "badcommand"로 변경하고 이를 적용한다.
+다음은 `kubectl describe pod lifecycle-demo` 를 실행하여 볼 수 있는 이벤트 출력 예시이다.
```
Comment (seokho-son) [inline content/ko/docs/concepts/containers/container-lifecycle-hooks.md:109]:
```suggestion
다음은 `kubectl describe pod lifecycle-demo` 실행을 통한 이벤트 출력 예시이다.
```
자연스럽게 변경해보았습니다.

**Review by seokho-son — COMMENTED** [review]
감사합니다. !

의견 하나 추가하였으며, 나머지는 다 좋아보입니다.

우선,
/lgtm

**Comment by jihoon-seo** [comment]
@seokho-son 리뷰 감사합니다. 해당 사항을 반영하고 squash까지 하였습니다. 확인 부탁드립니다. 😊

**Pull request merged**